apps/apps.c

```diff
@@ -2633,6 +2636,7 @@ void wait_for_async(SSL *s)
         fds++;
     }
     select(width, (void *)&asyncfds, NULL, NULL, NULL);
+    OPENSSL_free(origfds);
```

Comment:
I thought of this change too.  I think it's easier to make sense of but I wasn't sure it was worth the effort.  Given a second notice, I say go for it.